Source/FieldSolver/ElectrostaticSolver.cpp

```diff
@@ -411,7 +412,7 @@ WarpX::computePhiCartesian (const amrex::Vector<std::unique_ptr<amrex::MultiFab>
         else {
             AMREX_ALWAYS_ASSERT_WITH_MESSAGE(false,
                 "Field boundary conditions have to be either periodic or PEC "
-                "when using the electrostatic solver"
+                + "when using the electrostatic solver"
```

Comment:
This `+` is unnecessary and I think wrong. The C++ preprocessor will automatically concatenate adjacent string literals. Otherwise, you're doing pointer addition of the two resulting `const char*` values, which would crash or print gibberish while attempting to report the assertion failure. Try it and see?

Reply:
Thanks for the comment. I now tested it and it works fine with the `+` but it actually didn't achieve what I was hoping it would so I reverted the change. I've noticed that the error messages in  `AMREX_ALWAYS_ASSERT_WITH_MESSAGE` don't print as a single string when they are broken up over multiple lines such as the above case. For example triggering the above gives output:

```0::Assertion `false' failed, file "./Source/FieldSolver/ElectrostaticSolver.cpp", line 413, Msg: "Field boundary conditions have to be either periodic or PEC " + "when using the electrostatic solver" !!!```

and similarly without the `+` it gives:

```0::Assertion `false' failed, file "./Source/FieldSolver/ElectrostaticSolver.cpp", line 413, Msg: "Field boundary conditions have to be either periodic or PEC " "when using the electrostatic solver" !!!```